### PR TITLE
chore: added debug option in cliv2 Makefile

### PR DESCRIPTION
**Pull request by ShawkyZ**
## Pull Request Submission

Added an option to build the cliv2 go binary with flags for debugging and prevent symbol table stripping if DEBUG var is set to 1. 

Usage:
``DEBUG=1 make build`` 

The pull request must:

- **Reviewer Documentation**
    - [ ] follow [CONTRIBUTING](https://github.com/snyk/cli/blob/main/CONTRIBUTING.md) rules
    - [ ] be accompanied by a detailed description of the changes
    - [ ] contain a risk assessment of the change (Low | Medium | High) with regards to breaking existing functionality. A change e.g. of an underlying language plugin can completely break the functionality for that language, but appearing as only a version change in the dependencies.
    - [ ] highlight breaking API if applicable
    - [ ] contain a link to the automatic tests that cover the updated functionality.
    - [ ] contain testing instructions in case that the reviewer wants to manual verify as well, to add to the manual testing done by the author.
    - [ ] link to the link to the PR for the User-facing documentation
- **User facing Documentation**
    - [ ] update any relevant documentation in gitbook by submitting a gitbook PR, and including the PR link here
    - [ ] ensure that the message of the final single commit is descriptive and prefixed with either `feat:` or `fix:` , others might be used in rare occasions as well, if there is no need to document the changes in the release notes. The changes or fixes should be described in detail in the commit message for the changelog & release notes.
- **Testing**
    - [ ] Changes, removals and additions to functionality must be covered by acceptance / integration tests or smoke tests - either already existing ones, or new ones, created by the author of the PR.

## Pull Request Review

All pull requests must undergo a thorough review process before being merged. 
The review process of the code PR should include code review, testing, and any necessary feedback or revisions. 
Pull request reviews of functionality developed in other teams only review the given documentation and test reports. 

Manual testing will not be performed by the reviewing team, and is the responsibility of the author of the PR.

For Node projects: It’s important to make sure changes in `package.json` are also affecting `package-lock.json` correctly.

****************************If a dependency is not necessary, don’t add it.**************************** 

When adding a new package as a dependency, make sure that the change is absolutely necessary. We would like to refrain from adding new dependencies when possible.
Documentation PRs in gitbook are reviewed by Snyk's content team. They will also advise on the best phrasing and structuring if needed.

## Pull Request Approval

Once a pull request has been reviewed and all necessary revisions have been made, it is approved for merging into 
the main codebase. The merging of the code PR is performed by the code owners, the merging of the documentation PR 
by our content writers.

### CODE REVIEW

File: cliv2/Makefile [hunk 1]
```diff
@@ -149,7 +154,8 @@ configure: summary $(CACHE_DIR) $(CACHE_DIR)/variables.mk $(V1_DIRECTORY)/$(V1_E
 
 $(BUILD_DIR)/$(V2_EXECUTABLE_NAME): $(BUILD_DIR) $(SRCS) generate-ls-protocol-metadata
 	@echo "$(LOG_PREFIX) Building ( $(BUILD_DIR)/$(V2_EXECUTABLE_NAME) )"
-	@GOEXPERIMENT=$(FIPS_CRYPTO_BACKEND) GOOS=$(_GO_OS) GOARCH=$(GOARCH) $(GOCMD) build -tags=application -ldflags="$(LDFLAGS) -X github.com/snyk/snyk-ls/application/config.Version=$(LS_COMMIT_HASH) -X github.com/snyk/snyk-ls/application/config.LsProtocolVersion=$(LS_PROTOCOL_VERSION)  -X main.internalOS=$(GOOS) -X github.com/snyk/cli/cliv2/internal/embedded/cliv1.snykCLIVersion=$(CLI_V1_VERSION_TAG)" -o $(BUILD_DIR)/$(V2_EXECUTABLE_NAME) $(WORKING_DIR)/cmd/cliv2/*.go
+	@if [ "$(DEBUG)" = "1" ]; then echo "DEBUG is set to 1 LDFLAGS=$(LDFLAGS) GCFLAGS=$(GCFLAGS)"; fi
```
Comment (PeterSchafer) [inline cliv2/Makefile:157]: nitpick: Just to adapt the log output to the rest of the log messages.

```suggestion
	@echo "$(LOG_PREFIX)  DEBUG Build: $(DEBUG)"
	@echo "$(LOG_PREFIX)  LDFLAGS:     $(LDFLAGS)"
	@echo "$(LOG_PREFIX)  GCFLAGS:     $(GCFLAGS)"
```

File: cliv2/Makefile [hunk 2]
```diff
@@ -66,6 +66,11 @@ ifeq ($(_GO_OS), alpine)
 	_GO_OS = linux
 endif
 
+ifeq ($(DEBUG), 1)
```
Comment (PeterSchafer) [inline cliv2/Makefile:69]: Question: DEBUG is a very generic environment variable name. There are good chances to have this conflicting somewhere for example in our CI/CD. Should we maybe solve enabling the debug build via an explicit target?

**Comment by PeterSchafer** [comment]
Thanks @ShawkyZ for your contribution. I left you some comments, if you like to talk about the question, please let me know.

File: cliv2/Makefile (hunk 2 above)
Comment (ShawkyZ) [reply cliv2/Makefile:69]: It was going to be a bit of work to make a custom target. Maybe we can use other var that might not conflict? 
BUILD_DEBUG as an example ?

**Review by PeterSchafer — APPROVED** [review]
Praise: Very nice! Thanks for this improvements!

**Pull request merged**